src/game/client/snow.cpp

```diff
@@ -0,0 +1,78 @@
+/**
+ * @file
+ *
+ * @author tomsons26
+ *
+ * @brief Implements interface for drawing snow/rain effects.
+ *
+ * @copyright Thyme is free software: you can redistribute it and/or
+ *            modify it under the terms of the GNU General Public License
+ *            as published by the Free Software Foundation, either version
+ *            2 of the License, or (at your option) any later version.
+ *            A full copy of the GNU General Public License can be found in
+ *            LICENSE
+ */
+
+#include "snow.h"
+#include "weather.h"
+
+SnowManager::SnowManager() :
+    m_dimensionsBuffer(nullptr),
+    m_unk0(0),
+    m_velocity(0),
+    m_unk1(0),
+    m_freqScaleX(0),
+    m_freqScaleY(0),
+    m_amplitude(0),
+    m_pointSize(0),
+    m_maxPointSize(0),
+    m_minPointSize(0),
+    m_quadSize(0),
+    m_boxDimensions(0),
+    m_boxDensity(0),
+    m_isVisible(true)
+{
+}
+
+SnowManager::~SnowManager() {}
+
+void SnowManager::Init()
+{
+    m_dimensionsBuffer = new float[DIMENSION_BUFFER_SIZE];
+    m_unk0 = 0.0f;
+    Update_INI_Settings();
+}
+
+void SnowManager::Update_INI_Settings()
+{
+    float *buffer = m_dimensionsBuffer;
+    int dimensions = (int)g_theWeatherSetting->m_snowBoxDimensions;
+    for (int i = 0; i < DIMENSION_BUFFER_RES; ++i) {
+        for (int j = 0; j < DIMENSION_BUFFER_RES; ++j) {
+            *buffer = (double)(rand() % dimensions);
+            ++buffer;
+        }
+    }
+
+    m_velocity = g_theWeatherSetting->m_snowVelocity;
+    m_freqScaleX = g_theWeatherSetting->m_snowFreqScaleX;
+    m_freqScaleY = g_theWeatherSetting->m_snowFreqScaleY;
+    m_amplitude = g_theWeatherSetting->m_snowAmplitude;
+    m_pointSize = g_theWeatherSetting->m_snowPointSize;
+    m_quadSize = g_theWeatherSetting->m_snowQuadSize;
+    m_boxDimensions = g_theWeatherSetting->m_snowBoxDimensions;
+    m_boxDensity = 1.0f / g_theWeatherSetting->m_snowBoxDensity;
+    m_maxPointSize = g_theWeatherSetting->m_snowMaxPointSize;
+    m_minPointSize = g_theWeatherSetting->m_snowMinPointSize;
+    m_unk1 = m_boxDimensions / m_velocity;
```

Comment:
Time = Distance / Velocity.
This is probably the lifetime of the snow particle.
Seems a bit dangerous not having a 0.0f check on the box dimensions